### PR TITLE
Create new test for covering Refactoring ->Rename feature for DotNet LS

### PR DESCRIPTION
**Pull request by musienko-maxim**
### What does this PR do?
* Implement functional test for checking renaming class in DotNet projects.
* Add new simple (hello world) DotNet project to project templates
* Add new methods and locators for performing renaming with Cshrp LS.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10195
@dmytro-ndp, @Ohrimenko1988, @SkorikSergey

### CODE REVIEW

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    projectExplorer.openItemByPath(PROJECT_NAME);
+    projectExplorer.openItemByPath(PATH_TO_DOTNET_FILE);
+    consoles.waitExpectedTextIntoConsole(FINISH_LANGUAGE_SERVER_INITIALIZATION_MESSAGE);
+  }
+
+  @Test
+  public void checkRenaming() {
+    String newClassName = "HelloWorld";
+    String textFragmentAfterRenaming = "public class HelloWorld";
+    projectExplorer.openItemByPath(PROJECT_NAME);
```
Comment (SkorikSergey) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java]: This project was already opened in setUp() method(line 71).

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java [hunk 2]
```diff
@@ -2211,4 +2217,18 @@ public void launchCommentCodeFeature() {
...
         .perform();
   }
+
+  /**
+   * wait renaming field in the Editor (usually it field is used by language servers) type new
+   * rename value and wait closing of the field
+   *
+   * @param renamevalue
+   */
+  public void doRenamingByLanguageServerField(String renamevalue) {
```
Comment (SkorikSergey) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java]: It maybe better to rename variable to 'renameValue'

**Review by Ohrimenko1988 — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    consoles.selectProcessByTabName("dev-machine");
+    editor.goToCursorPositionVisible(18, 18);
+    menu.runCommand(ASSISTANT, REFACTORING, LS_RENAME);
+    editor.doRenamingByLanguageServerField(newClassName);
+
+    try {
+      editor.waitTextIntoEditor(textFragmentAfterRenaming);
+    } catch (TimeoutException ex) {
+      // remove try-catch block after issue has been resolved
+      fail("Known issue https://github.com/eclipse/che/issues/10180", ex);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java]: Is it permanent failure? If so, it's better to express it in error message like the follow:
```
Known permanent failure https://github.com/eclipse/che/issues/10180
```

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java (hunk 1 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java]: But in 72 line i open the project file - not project

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  @Inject private Wizard wizard;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private TestCommandServiceClient testCommandServiceClient;
+  @Inject private CommandsPalette commandsPalette;
+  @Inject private Consoles consoles;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = getClass().getResource("/projects/CsharpHelloWorld");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java]: To be consistent: `CsharpHelloWorld` > `CSharpHelloWorld`

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpFileEditingTest.java [hunk 5]
```diff
@@ -9,8 +9,9 @@
  * Contributors:
  *   Red Hat, Inc. - initial API and implementation
  */
-package org.eclipse.che.selenium.languageserver;
+package org.eclipse.che.selenium.languageserver.Csharp;
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpFileEditingTest.java]: package name should be in lower case: `Csharp` > `csharp`

**Review by SkorikSergey — APPROVED** [review]

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java [hunk 6]
```diff
@@ -38,6 +38,7 @@
 import static org.eclipse.che.selenium.pageobject.CodenvyEditor.Locators.IMPLEMENTATION_CONTAINER;
 import static org.eclipse.che.selenium.pageobject.CodenvyEditor.Locators.ITEM_TAB_LIST;
 import static org.eclipse.che.selenium.pageobject.CodenvyEditor.Locators.JAVA_DOC_POPUP;
+import static org.eclipse.che.selenium.pageobject.CodenvyEditor.Locators.LANGUAGE_SERVER_RENFACTORING_RENAME_FIELD_CSS;
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java]: typo `RENFACTORING`

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java [hunk 7]
```diff
@@ -212,6 +213,7 @@ public CodenvyEditor(
     String HOVER_POPUP_XPATH =
         "//div[@class='textviewTooltip' and contains(@style,'visibility: visible')]";
     String AUTOCOMPLETE_PROPOSAL_DOC_ID = "gwt-debug-content-assistant-doc-popup";
+    String LANGUAGE_SERVER_RENFACTORING_RENAME_FIELD_CSS = "input.orionCodenvy";
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java]: typo `RENFACTORING`

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java [hunk 8]
```diff
@@ -2211,4 +2216,15 @@ public void launchCommentCodeFeature() {
         .keyUp(CONTROL)
         .perform();
   }
+
+  /**
+   * wait renaming field in the Editor (usually it field is used by language servers), type new
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java:2221]: What does `Language Server Field` mean? Did you mean `Code Assistant`?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java (hunk 7 above)
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java]: `input.orionCodenvy` style name looks too general IMHO

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java (hunk 8 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java:2221]: Actually we have 2 UI part for renaming:
1. For java
selection_055

2. For other languages
selection_056

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java (hunk 7 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java]: I agree but it is part of Orion component. With this general name of css locator is simple and reliable. If you have other idea for this - we will apply it

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java:2221]: Done

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/csharp/CSharpRenamingTest.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.eclipse.che.selenium.pageobject.intelligent.CommandsPalette;
+import org.openqa.selenium.TimeoutException;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class CSharpRenamingTest {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/csharp/CSharpRenamingTest.java]: According to name convention there should be `CSharpClassRenamingTest` test class name, IMHO.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java (hunk 8 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java:2221]: I see, thank you for the explanation.

**Comment by musienko-maxim** [comment]
I totally agree with you but at this moment C# language is very unstable. During the test we can get an error that was described in the issue: https://github.com/eclipse/che/issues/10151. This one totally breaks the test environment.  And all tests after that will be broken. In this case we cannot track if the known issue has been fixed or not (because the workspace does not work properly).  Also we won't know which feature works, because all tests related to the workspace will fail anyway. So it is temporary solution i think after we stabilize and fix known issues,  we can join this into common class

**Comment by dmytro-ndp** [comment]
Got it! Thank you for response.

**Review by dmytro-ndp — APPROVED** [review]

**Comment by musienko-maxim** [comment]
ci-build

**Pull request merged**